src/login/logind-dbus.c

```diff
@@ -2206,7 +2206,10 @@ static int verify_shutdown_creds(
                                 interactive ? POLKIT_ALLOW_INTERACTIVE : 0,
                                 &m->polkit_registry,
                                 error);
-                if (r < 0)
+                /* If we get -EALREADY, it means a polkit decision was made, but not for
+                 * this action in particular. Assuming we are blocked on inhibitors,
+                 * ignore that error and allow the decision to be revealed below. */
+                if (r < 0 && !(blocked && r == -EALREADY))
```

Comment:
hmm, any chance you can change the error code from EALREADY to EBUSY at the same time?

I initially suggested that on the review of the original PR, and I think that would make things clearer here. To me EBUSY means "sorry, this thing is in an unusable state now, not going to accept more requests, go away" while EALREADY means "your request is OK, but I am already doing something for the moment, please come back later". one is a more negative the other a more positive error code. I think this one here actually deserves the more negative outloook, we are not going to every be able to request a polkit authentication from the same bus call after all. (yes, which error code to use here has nothing to do with the bug you are fixing, but I think it makes it easier to follow).

Reply:
hmmm, your fix is probably OK, but it assumes that the bus_verify_polkit_async_full() call further down will definitely catch this error. 

I think that's OK to assume, but this is a security check, hence I'd prefer if this is encoded in an assert(). i.e. set a local boolean here, and then check after all polkit  calls are done that it is false with a simple assert(). after all, if this EBUSY path is hit we should never be able to execute the actual payload of this function.

also, please add a comment explaining this at the assert()

Reply:
Yeah this all makes sense to me. I will make those changes, and I can look around for other places where similar logic is needed.